### PR TITLE
Remove deprecated ParametricPulse classes

### LINKED ISSUE #10810: Remove code deprecated in 0.22 (released on October 13, 2022) [pulse]
There is code released 9 month ago that needs to be removed.
 - `qiskit/pulse/utils.py:113`
 - `qiskit/pulse/library/parametric_pulses.py:66`
 - `qiskit/pulse/library/parametric_pulses.py:135`
 - `qiskit/pulse/library/parametric_pulses.py:247`
 - `qiskit/pulse/library/parametric_pulses.py:417`
 - `qiskit/pulse/library/parametric_pulses.py:540`
  
This includes:
- [ ] Remove deprecated code and tests (if they only check the deprecation raise)
- [ ] tag the PR as [`Changelog: Removal`](https://github.com/Qiskit/qiskit/labels/Changelog%3A%20Removal)
- [ ] create a release note in the category `upgrade:`. If you can include an example with an alternative for user to migrate to the new code, as this change might break users code.

Related:
* https://github.com/Qiskit/qiskit/issues/10812
* https://github.com/Qiskit/qiskit/issues/10811
* https://github.com/Qiskit/qiskit/issues/10813

### PR DESCRIPTION
**Pull request by Ak-ash22**
### Summary
Fixes #10810 

### Details and comments

Removed deprecated code parametric_pulses.py and tests. I noticed a bug in the related test.... in the qiskit/test/python/pulse/test_pulse_lib.py, there is a test class named TestParametricPulses(..) but it calls all the Gaussian, Constant and other waveforms from the symbolic_pulses.py and not from the paramteric_pulses.py. So I changed the name of the test class to TestSymbolicPulses(..).
Removed the related code from other files where ParametricPulses class was imported and used.
Created a reno 'remove-deprecated-parametric-pulses-class-667e4b970e1163b3.yaml' for the issue.

This PR has been requested to tagged as [Changelog: Removal](https://github.com/Qiskit/qiskit/labels/Changelog%3A%20Removal) to reflect the removal deprecated components.

### CODE REVIEW

**Comment by Ak-ash22** [comment]
@1ucian0 Its been a while, can you help me get the review from the code owners?

File: qiskit/qobj/converters/pulse_instruction.py
Comment (rupeshknn) [inline qiskit/qobj/converters/pulse_instruction.py]: you don't need the square brackets here

**Review by rupeshknn — APPROVED** [review]
LGTM

**Comment by nkanazawa1989** [comment]
Hi @Ak-ash22 we had another PR but the author @TsafrirA suggested to merge yours. Can you resolve conflicts?

**Comment by Ak-ash22** [comment]
Yes sure!.. I'll resolve it

**Comment by Ak-ash22** [comment]
@1ucian0 @nkanazawa1989 I have resolved all the conflicts and submitted it for PR. Kindly, take a look and help me merge it. Thank You!

File: releasenotes/notes/releasenotes/notes/remove-deprecated-parametric-pulses-class-667e4b970e1163b3.yaml
Comment (1ucian0) [inline releasenotes/notes/releasenotes/notes/remove-deprecated-parametric-pulses-class-667e4b970e1163b3.yaml]:
```suggestion
    All classes in it were deprecated since Qiskit 0.39 (with qiskit-terra 0.22), released on Oct 2022.
```

File: releasenotes/notes/releasenotes/notes/remove-deprecated-parametric-pulses-class-667e4b970e1163b3.yaml
Comment (1ucian0) [inline releasenotes/notes/releasenotes/notes/remove-deprecated-parametric-pulses-class-667e4b970e1163b3.yaml]:
```suggestion
    The module ``ParametricPulse`` is removed.
```

File: qiskit/pulse/library/symbolic_pulses.py
Comment (1ucian0) [inline qiskit/pulse/library/symbolic_pulses.py:1468]: great catch fixing the comments!

File: releasenotes/notes/releasenotes/notes/remove-deprecated-parametric-pulses-class-667e4b970e1163b3.yaml
Comment (1ucian0) [inline releasenotes/notes/releasenotes/notes/remove-deprecated-parametric-pulses-class-667e4b970e1163b3.yaml]:
```suggestion
    Instead, use :class:`.SymbolicPulse` and check its documentation for details.
```

**Review by 1ucian0 — COMMENTED** [review]
Thanks a lot for your work @Ak-ash22  !  

it looks good to me in general (just some comments in the release note). However, it would be great if @nkanazawa1989 can have the final 👍 . 

(and thanks @rupeshknn for reviewing it ahead!)

**Comment by nkanazawa1989** [comment]
Thanks @Ak-ash22 this looks good to me!

**Review by nkanazawa1989 — APPROVED** [review]

**Pull request merged**
